core/domain/base_model_validators.py

```diff
@@ -84,12 +89,78 @@ def __init__(
             model_class: ClassObject. The external model class.
             model_ids: list(str). The list of external model ids to fetch the
                 external models.
-        """
+
+        Raises:
+            Exception. This class was used instead of
+                UserSettingsModelFetcherDetails for the UserSettingsModel.
+        """
+        if model_class == user_models.UserSettingsModel:
+            raise Exception(
+                'When fetching instances of UserSettingsModel, please use ' +
+                'UserSettingsModelFetcherDetails instead of ' +
+                'ExternalModelFetcherDetails')
         self.field_name = field_name
         self.model_class = model_class
         self.model_ids = model_ids
 
 
+class UserSettingsModelFetcherDetails(python_utils.OBJECT):
+    """Value object providing ids to fetch the user settings model."""
+
+    def __init__(
+            self, field_name, model_ids,
+            may_contain_system_ids=False, may_contain_pseudonymous_ids=False):
+        """Initializes the UserSettingsModelFetcherDetails domain object.
+
+        Args:
+            field_name: str. A specific name used as an identifier by the
+                storage model which is used to identify the user settings model
+                reference. For example: 'committer_id': UserSettingsModel,
+                committer_id is the field name to identify the external model
+                UserSettingsModel.
+            model_ids: list(str). The list of user settings model ids to fetch
+                the UserSettingsModels.
+            may_contain_system_ids: bool. Whether the model ids contain
+                system ids which should be omitted before attempting to fetch
+                the corresponding models.
+            may_contain_pseudonymous_ids: bool. Whether the model ids contain
+                pseudonymous ids which should be omitted before attempting to
+                fetch the corresponding models.
+
+                Note:
+                    Set may_contain_pseudonymous_ids to True if and only if
+                    this field can contain user IDs that are pseudonymized as
+                    part of Wipeout. These fields can only be in models that
+                    have LOCALLY_PSEUDONYMIZE as their DELETION_POLICY. Set
+                    may_contain_system_ids to True if and only if this field
+                    can contain admin or bot IDs.
+        """
+        filtered_model_ids = model_ids
+        if may_contain_system_ids:
+            filtered_model_ids = list(
+                set(filtered_model_ids) - set(feconf.SYSTEM_USERS.values()))
+        else:
+            if set(filtered_model_ids) & set(feconf.SYSTEM_USERS.values()):
+                raise Exception(
+                    'The attribute may_contain_system_ids is set to False ' +
+                    'but the model_ids contain system ids.')
```

Comment:
I think we cannot produce typical error here, we will need some kind of error that is then catched by code that transforms it to error (`BaseModelValidator._add_error`). Otherwise this would mean that when we run the jobs on production it would not put the error into the output but the job would produce a lot of exceptions and fail to complete.

Reply:
The catch should be done in `BaseModelValidator._fetch_field_name_to_external_model_references`

Reply:
I think the job is expected to fail to complete because it is an internal error, rather than a data validation error. So if we use _add_error won't it continue running and eventually the error pops up when the user setting model is not able to fetch the system id.

Reply:
Hmm, I see, although I still think that the error is not internal because it is not a badly written job when this error occurs (the error can be caused by someone adding a wrong IDs in the field). Could this be written in a way that we still have it reported ass validation error and also the fetch is not done if this error occurred?

Reply:
I have added the error using _add_error. Please check. Thanks